### DISCUSSION TITLE
Custom Default Filename

### DISCUSSION THREAD

**Discussion by dvdfu** [yasirkula/UnityNativeFilePicker#8] [Ideas]
Hi, first off I love this plugin! It's really useful for me and has a friendly API.

I'd love if the `ExportFile` method included an argument that allows the file path (used to locate the file to export) to differ from the file name that appears by default for the user when they save the file.

My specific use case is that I use Unity's `temporaryCachePath` to prepare files (audio recordings) right before they're exported by this plugin, and I want the exported file name to include a timestamp of the file creation. If these filenames need to match, then the number of different files in the temporary cache path can grow to be quite large!

**Comment by yasirkula** [comment]
I've googled the issue for a bit. I couldn't find a way to set a custom filename on iOS, UIDocumentPickerViewController doesn't seem to support it. How about this solution:

- Rename file via File.Move to add timestamp
- Call ExportFile
- Rename file via File.Move to the old name inside FilesExportedCallback
